Richb css

richb-hanover opened the pull request:
This is now good enough to criticize:

* It matches the styles of the forum & wiki:
  * Logo size is the same as the forum (I think we can adjust the wiki if we like this appearance)
  * Uses same font & size as the wiki
  * Uses the blue color from the logo as the link color
* `test.sh` is a script that uses `dir-index.cgi` to generate the four variations of the page
* README.md tells stuff about building and testing
* All files rely on `/static/` at the root of the server to contain the CSS file and logo file
* Removed (most) references to LEDE

It's a first cut - please let's keep working on it. Thanks.

aparcar commented:
Hi, thanks for your work, some requests:

* Please rebase this, `style.css` got some issues.
* Please copy all test data into the test folder, since all assets are linked with absolute path and you're using a web server, it should work as expected.
* Please add test/ to .gitignore
* Please don't use the full width for tables but keep it centered in the middle, the current design looks silly on widescreen.
![image](https://user-images.githubusercontent.com/16000931/133854068-4968e94c-6ccc-42ff-a4bc-dfdbd50ef90b.png)
* In my currently deployed version I moved the logo to be centered as well, could you please do that too?
![image](https://user-images.githubusercontent.com/16000931/133854136-44d74e66-d131-4880-a9fa-a2c1d3a99979.png)

richb-hanover commented:
Feedback! This is great. I will work on this over the weekend.

richb-hanover commented:
OK. Here's a new "good version" to test:

* Rebased against current `little-steps`
* We can continue to harmonize the Download, Wiki, and Forum styling
* All sample files are in `/test` and I removed old, crufty data from earlier versions
* Tables are now "as wide as they need to be, and no wider" (removed `width: 1%` that was causing problems.)
* I did *not* center the contents in the middle of the page. 1) It would be a lot of work to make the logo move around to the match the page contents. `index.html` is pretty narrow, but the `targets.html` is extremely wide. 2) I was not fond of having the OpenWrt logo move around the header depending on which page was shown.

**Question about link color**

* The OpenWrt wiki uses a link color of `#337ab7`;
* The link color on these pages is currently  `#00B5E2`; as specified in the OpenWrt Logo Guidelines (now included in the repo)
* *BUT...* the actual color of the bright blue circle of the logo is `#43b0e8`. They look similar, but not identical. How can this be resolved?

aparcar commented:
> We can continue to harmonize the Download, Wiki, and Forum styling

Agree, let's not stall this much longer :)

> I did not center the contents in the middle of the page. 1) It would be a lot of work to make the logo move around to the match the page contents. index.html is pretty narrow, but the targets.html is extremely wide. 2) I was not fond of having the OpenWrt logo move around the header depending on which page was shown.

Fine, let's keep the logo on the left but please keep the content centered, I think it looks worse then before having all on the left. There must be a CSS way to tell both tables to be "as small as possible but as wide as necessary". The wider table should determine how wide the smaller table is. @mwarning ideas?

 > BUT... the actual color of the bright blue circle of the logo is #43b0e8. They look similar, but not identical. How can this be resolved?

This seems to be a error in the style guide then, yikes. I guess we need a v2. @ynezz can you have a look at this and talk to the designer?

richb-hanover commented:
Thanks. @aparcar @mwarning I tweaked the index.html page to use flexbox for each of the 'kinds' of files. That solves some of the "skinny page" ("all on the left") problems. (I don't think a simple directory listing is seen very often...)

re: Proper color for use in the logo. Yes, let's ask the designer.

re: Link color. Maybe we can get advice from the designer as well?

aparcar commented:
I don't see a difference since using the flexboxes on the target page:

![image](https://user-images.githubusercontent.com/16000931/133923849-dba3022d-9deb-4359-a8b3-4b1eb907f47d.png)

The overview page is fine-ish but if we go this route we should have the archive below (like my example in the forum)
![image](https://user-images.githubusercontent.com/16000931/133923877-57626157-c41a-46fb-b3a3-3ca72e54e0ae.png)

richb-hanover commented:
It feels to me as if we're working on two parallel tracks, but without any obvious place to sync up.

* We're planning (I think) to "put the OFS on top" of the index.html at downloads.openwrt.org. I strongly support this plan
* I've been tweaking the `little-steps` branch to make it look more like the Wiki.

I compared the CSS between the two branches. They're similar, but neither is a drop-in replacement for the other. Before we do a lot more work, I want to be sure we have a shared goal for how each page looks. Can we agree on this?

* All pages: Black header, OpenWrt logo (~3em) at the left. The d.o.o page with the OFS will have a language selection in the header at the right side. All content will be "as wide as it needs to be", centered in the page.

* d.o.o page will initially show with the OFS, substantially as it does in firmware-selector.openwrt.org. (It's centered on the page, ~600px wide.) Clicking the "All Downloads" will un-hide the list of other downloadable files, either as a list (as in the current d.o.o page, or as blocks for various kinds of files (Release, Previous, Snapshots, Archived, and RC versions when they're available). 

* (@aparcar - I realize that my latest prototype for index.html was a pale imitation of your (sexier) earlier version. You were right about using blocks to represent the various kinds of downloads. My biggest remaining concern is the coloring (green/yellow/red) for the blocks - I think we should find a neutral color for the headers within each block: perhaps a nice grey, or perhaps the blue color from the logo...)

* Each "targets" page is inherently a wide layout - the tables extend pretty far side-to-side. The tables should only be as wide as necessary, and white space should grow equally on the sides.

* The "archive" page will be centered, as a long list of the links to the previous releases.
  
* The 404 page should center its text between the white space of the margins. If viewed on a very wide screen, it might look a bit narrow, but people should (almost) never see that page.

* The "Directory" page's contents will also be centered, and will similarly appear quite narrow. If I understand correctly, that page will be relatively infrequent.

* The final JSON page won't be viewed directly. CSS has no effect on it.

aparcar commented:
> It feels to me as if we're working on two parallel tracks, but without any obvious place to sync up.
> 
>     * We're planning (I think) to "put the OFS on top" of the index.html at downloads.openwrt.org. I strongly support this plan

Yes, this is part 2 in my plan. First adopt the style, then integrate OFS. @mwarning is still fine tuning OFS which may take another week or two, I don't want to make the re-style depend on that.

> 
>     * I've been tweaking the `little-steps` branch to make it look more like the Wiki.
> 
Which is great! It already looks very similar which is exactly what we planed to do!
> 
> I compared the CSS between the two branches. They're similar, but neither is a drop-in replacement for the other. Before we do a lot more work, I want to be sure we have a shared goal for how each page looks. Can we agree on this?
> 
>     * All pages: Black header, OpenWrt logo (~3em) at the left. The d.o.o page with the OFS will have a language selection in the header at the right side. All content will be "as wide as it needs to be", centered in the page.
:heavy_check_mark: 
> 
>     * d.o.o page will initially show with the OFS, substantially as it does in firmware-selector.openwrt.org. (It's centered on the page, ~600px wide.) Clicking the "All Downloads" will un-hide the list of other downloadable files, either as a list (as in the current d.o.o page, or as blocks for various kinds of files (Release, Previous, Snapshots, Archived, and RC versions when they're available).
:heavy_check_mark: 
> 
>     * (@aparcar - I realize that my latest prototype for index.html was a pale imitation of your (sexier) earlier version. You were right about using blocks to represent the various kinds of downloads. My biggest remaining concern is the coloring (green/yellow/red) for the blocks - I think we should find a neutral color for the headers within each block: perhaps a nice grey, or perhaps the blue color from the logo...)
> 
>     * Each "targets" page is inherently a wide layout - the tables extend pretty far side-to-side. The tables should only be as wide as necessary, and white space should grow equally on the sides.
Yes, ideally we make both table the same with (if that's possible with CSS). Else it looks clumsy where the lower table is wider than the upper one.
> 
>     * The "archive" page will be centered, as a long list of the links to the previous releases.
:heavy_check_mark: 
> 
>     * The 404 page should center its text between the white space of the margins. If viewed on a very wide screen, it might look a bit narrow, but people should (almost) never see that page.
:heavy_check_mark: 
> 
>     * The "Directory" page's contents will also be centered, and will similarly appear quite narrow. If I understand correctly, that page will be relatively infrequent.
We could give it a minimum width like it's done right now. I wouldn't shrink it too much.
> 
>     * The final JSON page won't be viewed directly. CSS has no effect on it.
Yes, even thought JSON might be extended in some way or another, but that's TBD somewhere else.

aparcar commented:
@richb-hanover could you please apply the [changes](https://github.com/aparcar/openwrt-download-page/pull/1#pullrequestreview-760953869) suggested by @jow-?

richb-hanover commented:
I applied @jow changes, except for incorporating the SVG into the page. I think that change deserves a site-wide consideration of how to include that image. (It's a PNG in both the wiki and the forum.)

I also restored the `index.html` page back to listing the four download types top-to-bottom (instead of blocks - they're commented out)

I think this is good enough to continue the process of putting the OFS on top of the (restored) `index.html`page, which would get us to a point that it could become downloads.openwrt.org... True?
